userbenchmark/optim/run.py

```diff
@@ -405,33 +406,48 @@ def run_model(modelName, device, Optim, defaults, maybe_pt2_):
 
         print(f'{datetime.datetime.now()}     python -m userbenchmark.optim.run -m {modelName} -d {device}' +
               f' -o {Optim.__name__} --df "{defaults_to_str(defaults)}" -f {maybe_pt2_}')
+
+        compile_r = None
+        sub_label = f'{modelName}, {optim.__class__.__name__}, {device}'
+        description = pt2_description + defaults_to_str(defaults)
+
+        # Get compile time by running 5 times and subtracting
+        #   first entry - avg(entries 3 through 5)
+        # skipping the second entry due to high variance of first cache hit
+        if maybe_pt2_ != '':
+            times = []
+            if device == "cuda":
+                torch.cuda.reset_peak_memory_stats()
+                torch.cuda.empty_cache()
+            for _ in range(5):
+                t0 = time.perf_counter()
+                pt2_optimizer_step(optim)
+                t1 = time.perf_counter()
+                times.append(t1 - t0)
+            compile_r = (f'compile_time, {sub_label}, {description}', times[0] - sum(times[2:5]) / 3)
```

Comment:
In torchbench framework, we measure the compile time by calling this API: https://github.com/pytorch/benchmark/blob/main/torchbenchmark/util/model.py#L358